images/macos/provision/utils/xcode-utils.sh

```diff
@@ -68,7 +68,7 @@ extractXcodeXip() {
     local XCODE_VERSION="$2"
     XCODE_XIP="${WORKING_DIR}/Xcode_${XCODE_VERSION// /_}.xip"
 
-    open -W $XCODE_XIP
+    cd "${WORKING_DIR}" && xip -x "${XCODE_XIP}"
```

Comment:
I suggest using 
```
push $WORKING_DIR
xip -x "${XCODE_XIP}"
popd
```

or move `cd` to `xcode-tools.sh` level. Your function mustn't have side effects like switching current directory